### PR TITLE
feat(test): Add e2e tests for http rest connector

### PR DESCRIPTION
**Pull request by sbuettner**
## Related issues

closes https://github.com/camunda/team-connectors/issues/532

### CODE REVIEW

File: connectors-e2e-test/connector-http-json-test/pom.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  <parent>
+    <groupId>io.camunda.connector</groupId>
+    <artifactId>connector-parent</artifactId>
+    <relativePath>../../parent/pom.xml</relativePath>
+    <version>8.4.0-SNAPSHOT</version>
+  </parent>
+
+  <description>Tests</description>
+  <artifactId>connector-http-json-test</artifactId>
+  <packaging>jar</packaging>
```
Comment (chillleader) [inline connectors-e2e-test/connector-http-json-test/pom.xml:14]: Will we need this module packaged as a jar for anything apart from our CI/CD pipelines? If not, then I would suggest disabling `maven-shade-plugin` for it (originally defined in the parent) to save some build time.

File: connectors-e2e-test/connector-http-json-test/src/test/java/io/camunda/connector/e2e/rest/HttpJsonConnectorTest.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    classes = {TestConnectorRuntimeApplication.class},
+    properties = {
+      "spring.main.allow-bean-definition-overriding=true",
+      "camunda.connector.webhook.enabled=false",
+      "camunda.connector.polling.enabled=false"
+    })
+@ZeebeSpringTest
+@ExtendWith(MockitoExtension.class)
+// TODO discuss whether we want to enable it by default
+// @EnabledIfEnvironmentVariable(named = "CONNECTORS_E2E_TEST", matches = "true")
```
Comment (chillleader) [inline connectors-e2e-test/connector-http-json-test/src/test/java/io/camunda/connector/e2e/rest/HttpJsonConnectorTest.java:60]: Yes, that would be great 👍  We could then add a new pipeline for running e2e tests or add a checkbox to optionally run it during snapshot / release builds.

**Review by chillleader — COMMENTED** [review]
This looks great!

File: connectors-e2e-test/connector-http-json-test/pom.xml (hunk 1 above)
Comment (sbuettner) [reply connectors-e2e-test/connector-http-json-test/pom.xml:14]: The `maven-shade-plugin` is configured for the connectors. The tests project is not inheriting from the connectors pom and therefore the plugin is not enabled.

**Pull request merged**